presto-spi/src/main/java/com/facebook/presto/spi/block/AbstractSingleRowBlock.java

```diff
@@ -21,137 +21,135 @@
 public abstract class AbstractSingleRowBlock
         implements Block
 {
-    // in AbstractSingleRowBlock, offset is position-based (consider as cell-based), not entry-based.
-    protected final int startOffset;
+    protected final int rowOffset;
 
-    protected final int numFields;
+    protected AbstractSingleRowBlock(int rowOffset)
+    {
+        this.rowOffset = rowOffset;
+    }
 
     protected abstract Block getFieldBlock(int fieldIndex);
 
-    protected AbstractSingleRowBlock(int startOffset, int numFields)
-    {
-        this.startOffset = startOffset;
-        this.numFields = numFields;
-    }
+    @Override
+    public abstract int getPositionCount();
```

Comment:
This abstract method definition is redundant  (it's defined by `Block` interface)